### PR TITLE
fix: handle uri in object

### LINKED ISSUE #1078: bug(node): error when using local storage
### LanceDB version

v0.4.11

### What happened?

In short, I called

```
lancedb.connect({
  "uri": "/var/folders/j5/zqtb_vtx6p166jkfgpkcrfd80000gp/T/lance-db-test-hoMBti"
})
```

Then I got an exception:

```
Error: Invalid credentials configuration
    at LocalConnection.createTableImpl (...node_modules/vectordb/src/index.ts:715:35)
    at LocalConnection.createTable (...node_modules/vectordb/src/index.ts:676:12)
```

things seem to work if I do this:

```
lancedb.connect({
  awsCredentials: undefined,
  awsRegion: undefined,
  "uri": "/var/folders/j5/zqtb_vtx6p166jkfgpkcrfd80000gp/T/lance-db-test-hoMBti"
})
```

### Are there known steps to reproduce?

_No response_

### PR DESCRIPTION
**Pull request by wjones127**
Fixes #1078

### CODE REVIEW

File: node/src/index.ts [hunk 1]
```diff
@@ -176,16 +176,21 @@ export async function connect (
...
+    } else {
+      opts = Object.assign(
+        {
+          uri: '',
+          awsCredentials: undefined,
+          awsRegion: defaultAwsRegion,
+          apiKey: undefined,
+          region: defaultAwsRegion
+        },
+        arg
```
Comment (flaviut) [inline node/src/index.ts:191]: you're obviously more familiar with this code than I am, but I would have thought that the best solution here would be to explicitly check the URL type & follow two different code paths, like `RemoteConnection` below.

**Review by flaviut — APPROVED** [review]
this is extremely similar to in effect to my personal workaround -- should work great.

File: node/src/index.ts (hunk 1 above)
Comment (flaviut) [reply node/src/index.ts:191]: (or [perhaps edit the rust part to not try and validate aws creds](https://github.com/lancedb/lancedb/blob/main/rust/ffi/node/src/lib.rs#L205-L240) when it doesn't receive an S3 url)

Comment (wjones127) [reply node/src/index.ts:191]: TBH we are a few weeks away from stopping development of this SDK in favor of the new Node one, so I'm not inclined to put too much effort here. This seemed like the easiest quick fix.

**Review by westonpace — APPROVED** [review]

**Pull request merged**